cpp/src/arrow/flight/client.cc

```diff
@@ -90,9 +90,6 @@ std::shared_ptr<FlightWriteSizeStatusDetail> FlightWriteSizeStatusDetail::Unwrap
   return std::dynamic_pointer_cast<FlightWriteSizeStatusDetail>(status.detail());
 }
 
-FlightClientOptions::FlightClientOptions()
-    : write_size_limit_bytes(0), disable_server_verification(false) {}
-
 FlightClientOptions FlightClientOptions::Defaults() { return FlightClientOptions(); }
```

Comment:
Should this be restored?

Reply:
I don't think so, since the defaults are set with field initializers - no need to also make an explicit constructor and duplicate things.

Reply:
Hmm... a bit lost about where the the fields are explicitly initialized.
Do you mean depends on struct zero initialization, as the two fields happen to be zero?
I don't see initial values in FlightClientOptions declaration https://github.com/apache/arrow/blob/master/cpp/src/arrow/flight/client.h#L116

Reply:
Sorry, I think I did something weird when rebasing. It's in the PR now.